oak-search-elastic/src/test/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticDynamicBoostQueryTest.java

```diff
@@ -0,0 +1,143 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.plugins.index.elastic;
+
+import org.apache.jackrabbit.JcrConstants;
+import org.apache.jackrabbit.oak.api.CommitFailedException;
+import org.apache.jackrabbit.oak.api.Tree;
+import org.apache.jackrabbit.oak.api.Type;
+import org.apache.jackrabbit.oak.plugins.index.search.util.IndexDefinitionBuilder;
+import org.apache.jackrabbit.oak.plugins.nodetype.write.NodeTypeRegistry;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.UUID;
+
+public class ElasticDynamicBoostQueryTest extends ElasticAbstractQueryTest {
+
+    public static final String ASSET_NODE_TYPE =
+            "[dam:Asset]\n" +
+                    " - * (UNDEFINED) multiple\n" +
+                    " - * (UNDEFINED)\n" +
+                    " + * (nt:base) = oak:TestNode VERSION";
+
+    private static final String UNSTRUCTURED = "nt:unstructured";
+
+    @Test
+    public void dynamicBoost() throws CommitFailedException {
+        configureIndex();
+
+        Tree test = createNodeWithType(root.getTree("/"), "test", UNSTRUCTURED);
+        Tree item1Metadata = createNodeWithMetadata(test, "item1", "flower with a lot of red and a bit of blue");
+        Tree item1Color1 = createNodeWithType(item1Metadata,"color1", UNSTRUCTURED);
+        item1Color1.setProperty("name", "red");
+        item1Color1.setProperty("confidence", 9.0);
+        Tree item1Color2 = createNodeWithType(item1Metadata,"color2", UNSTRUCTURED);
+        item1Color2.setProperty("name", "blue");
+        item1Color2.setProperty("confidence", 1.0);
+
+        Tree item2Metadata = createNodeWithMetadata(test, "item2", "flower with a lot of blue and a bit of red");
+        Tree item2Color1 = createNodeWithType(item2Metadata,"color1", UNSTRUCTURED);
+        item2Color1.setProperty("name", "blue");
+        item2Color1.setProperty("confidence", 9.0);
+        Tree item2Color2 = createNodeWithType(item2Metadata,"color2", UNSTRUCTURED);
+        item2Color2.setProperty("name", "red");
+        item2Color2.setProperty("confidence", 1.0);
+        root.commit();
+
+        assertEventually(() -> {
+            assertQuery("//element(*, dam:Asset)[jcr:contains(@title, 'flower')]",
+                    XPATH, Arrays.asList("/test/item1", "/test/item2"));
+            assertOrderedQuery("select [jcr:path] from [dam:Asset] where contains(title, 'red flower')",
+                    Arrays.asList("/test/item1", "/test/item2"));
+            assertOrderedQuery("select [jcr:path] from [dam:Asset] where contains(title, 'blue flower')",
+                    Arrays.asList("/test/item2", "/test/item1"));
+        });
+    }
+
+    @Test
+    public void dynamicBoostAnalyzed() throws CommitFailedException {
```

Comment:
This test differs from the previous test in just the search terms - "red flower" vs "red-flower".  IIUC both the tests used analyzed fields and queries. Could you please explain the purpose of this test?

Reply:
The `dynamicBoost` test queries the value using an input text that contains the same token (`red`) present in the dynamic tags. Instead `dynamicBoostAnalyzed` uses a text (red-flower) that without analysis would not match the tags indexed by the dynamic boost function (this happens for example in lucene). Thanks to the fact the input gets analyzed the tokens produced are `red` and `flower`.

Reply:
Okay. The way I was looking at it is - query string "red flower" gets tokenized into "red" and "flower" as does the query string - "red-flower". 

So, is the intention here to test that query string is being tokenized properly?